File: app/src/main/java/com/coinninja/coinkeeper/model/db/converter/AccountStatusConverter.java
package com.coinninja.coinkeeper.model.db.converter;

import com.coinninja.coinkeeper.model.db.enums.AccountStatus;

import org.greenrobot.greendao.converter.PropertyConverter;

public class AccountStatusConverter implements PropertyConverter<AccountStatus, Integer> {
    @Override
    public AccountStatus convertToEntityProperty(Integer databaseValue) {
        if (databaseValue == null) {
            return null;
        }

        for (AccountStatus status : AccountStatus.values()) {
            if (status.getId() == databaseValue) {
                return status;
            }
        }

        return AccountStatus.UNVERIFIED;
    }

    @Override
    public Integer convertToDatabaseValue(AccountStatus status) {
        return status == null ? null : status.getId();
    }
}

File: persistance/src/main/java/app/coinninja/cn/persistance/converter/AccountStatusConverter.kt
package app.coinninja.cn.persistance.converter

import androidx.room.TypeConverter
import app.coinninja.cn.persistance.model.AccountStatus

object AccountStatusConverter {
    @TypeConverter
    @JvmStatic
    fun to(id: Int?): AccountStatus? = id?.let { return AccountStatus.from(it) }

    @TypeConverter
    @JvmStatic
    fun from(status: AccountStatus?): Int? = status?.let { return it.id }
}